src/Services/FHIR/FhirImmunizationService.php

```diff
@@ -0,0 +1,217 @@
+
+        $immunizationResults = sqlStatement($SQL, $search['patient']);
+        $results = array();
+        while ($row = sqlFetchArray($immunizationResults)) {
+            array_push($results, $row);
+        }
+        return $results;
+    }
+
+    public function getOne($id)
+    {
+        $SQL = "SELECT id,
+                    patient_id,
+                    administered_date,
+                    cvx_code,
+                    manufacturer,
+                    lot_number,
+                    administered_by_id,
+                    administered_by,
+                    education_date,
+                    note,
+                    create_date,
+                    amount_administered,
+                    amount_administered_unit,
+                    expiration_date,
+                    route,
+                    administration_site,
+                    completion_status,
+                    refusal_reason 
+                    FROM Immunizations
+                    WHERE id = ?";
+
+        $sqlResult = sqlStatement($SQL, $id);
+        $result = sqlFetchArray($sqlResult);
+
+        return $result;        
+    }
+
+    public function createImmunizationResource($id = '', $data = '', $encode = true)
+    {     
+        $status = new FHIRImmunizationStatusCodes();
+        $statusCoding = new FHIRCoding();
+        if (isset($data['completion_status'])) {
+            $statusCoding = new FHIRCoding();
+            $statusCoding->setSystem('http://hl7.org/fhir/event-status');
+            $statusCoding->setCode($data['completion_status']);
+            $statusCoding->setDisplay($data['completion_status']);
+            $status->setValue($statusCoding);
+        }
+        else {
+            $statusCoding->setSystem("http://terminology.hl7.org/CodeSystem/data-absent-reason");
+            $statusCoding->setCode("unknown");
+            $statusCoding->setDisplay("Unknown");        
+            $status->setValue($statusCoding);
+        }
+        
+        $statusReason = new FHIRCodeableConcept();
+        $statusReasonCoding = new FHIRCoding();
+        $statusReasonCoding->setSystem("http://terminology.hl7.org/CodeSystem/v3-ActReason");
+        $statusToCodeMapping = array(
+                                    "immunity" => "IMMUNE",
+                                    "medical precaution" => "MEDPREC",
+                                    "product out of stock" => "OSTOCK",
+                                    "patient objection" => "PATOBJ"
+                                );
+        $statusReasonCoding->setCode($statusToDisplayMapping[strtolower($data['refusal_reason'])]);
+        $statusReasonCoding->setDisplay($data['refusal_reason']);
+        $statusReason->addCoding($statusReasonCoding);
+        
+        $vaccineCode = new FHIRCodeableConcept();
+        $vaccineCodeCoding = new FHIRCoding();
+        $vaccineCodeCoding->setSystem("http://hl7.org/fhir/sid/cvx");
```

Comment:
This url seems broken.

Reply:
I'm not worried if valid url to terminology build for now.